core/store/src/trie/mem/updating.rs

```diff
@@ -831,10 +817,26 @@ impl<'a, M: ArenaMemory> MemTrieUpdate<'a, M> {
 
         // We may need to change node type to keep the trie structure unique.
         for node_id in path.into_iter().rev() {
-            self.squash_node(node_id);
+            self.squash_node(node_id).unwrap();
         }
     }
+}
 
+impl<
```

Comment:
Why not just impl the GenericTrieUpdate directly?

Reply:
This is because otherwise, `MemTrieUpdate` and `NodesStorage` don't see `squash_node` implementation.
And if I do direct conversion of `self` to `dyn GenericTrieUpdate<'a, MemTrieNodeId, FlatStateValue>`, I get errors of the kind

```
lifetime may not live long enough
requirement occurs because of a mutable reference to `dyn GenericTrieUpdate<'_, MemTrieNodeId, FlatStateValue>`
```

Looks like something about lifetimes for templates, and this separate trait does the trick.